dapp-core/src/main/kotlin/jp/co/soramitsu/dapp/service/DappService.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jp.co.soramitsu.dapp.service
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import io.reactivex.schedulers.Schedulers
+import jp.co.soramitsu.dapp.AbstractDappScript
+import jp.co.soramitsu.dapp.helper.CacheManager
+import jp.co.soramitsu.dapp.parser.ContractParser.Companion.parse
+import jp.co.soramitsu.iroha.java.IrohaAPI
+import jp.co.soramitsu.iroha.java.Utils
+import mu.KLogging
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import java.security.KeyPair
+import java.util.concurrent.Executors
+
+@Component
+class DappService(
+    @Autowired
+    private val irohaAPI: IrohaAPI,
+    @Autowired
+    private val dappKeyPair: KeyPair,
+    @Autowired
+    private val observableSource: CommandObservableSource,
+    @Autowired
+    private val cacheManager: CacheManager,
+    @Autowired
+    private val contractsRepositoryMonitor: ContractsRepositoryMonitor
+) {
+    private val scheduler = Schedulers.from(
+        Executors.newFixedThreadPool(
+            2,
```

Comment:
Why 2?

Reply:
one is for enabled contract, other is for disabled, can be one still, but I suppose two is more describing

Reply:
btw changed to `notary-commons`